File: articles/algorithm-design-principles/java/src/main/java/org/contentcatalyst/algorithms/AlgorithmDesignAudit.java
package org.contentcatalyst.algorithms;
public class AlgorithmDesignAudit {
  static boolean nondecreasing(int[] xs){ for(int i=0;i<xs.length-1;i++){ if(xs[i] > xs[i+1]) return false; } return true; }
  public static void main(String[] args){
    System.out.println("test_name,status");
    System.out.println("sorted_valid," + (nondecreasing(new int[]{1,2,2,3}) ? "pass" : "fail"));
    System.out.println("sorted_invalid," + (nondecreasing(new int[]{1,3,2}) ? "pass" : "fail"));
  }
}

File: articles/algorithm-design-principles/c/algorithm_design_audit.c
#include <stdio.h>
int nondecreasing(int *xs, int n){ for(int i=0;i<n-1;i++){ if(xs[i] > xs[i+1]) return 0; } return 1; }
int main(void){ int a[]={1,2,2,3}; int b[]={1,3,2}; printf("test_name,status\nsorted_valid,%s\nsorted_invalid,%s\n", nondecreasing(a,4)?"pass":"fail", nondecreasing(b,3)?"pass":"fail"); return 0; }
